server.go

```diff
@@ -182,16 +182,22 @@ func (s *server) handlePacket(pconn net.PacketConn, remoteAddr net.Addr, packet
 			version,
 			hdr.ConnectionID,
 			s.scfg,
-			s.closeCallback,
 			s.cryptoChangeCallback,
 		)
 		if err != nil {
 			return err
 		}
-		go session.run()
 		s.sessionsMutex.Lock()
 		s.sessions[hdr.ConnectionID] = session
 		s.sessionsMutex.Unlock()
+
+		go func(id protocol.ConnectionID) {
+			// session.run() returns as soon as the session is closed
+			_ = session.run()
+
+			s.removeConnection(id)
+		}(hdr.ConnectionID)
```

Comment:
hdr is not in a loop, so it should be ok to just capture hdr.ConnectionID, right?